engine/flink/model-util/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/AggregatorFunction.scala

```diff
@@ -84,7 +87,11 @@ trait AggregatorFunctionMixin { self: StateHolder[TreeMap[Long, AnyRef]] =>
     (timestamp / minimalResolutionMs) * minimalResolutionMs
   }
 
-  protected def stateForTimestamp[T](stateValue: TreeMap[Long, T], timestamp: Long): TreeMap[Long, T] = {
+  protected def stateForTimestampToSave[T](stateValue: TreeMap[Long, T], timestamp: Long): TreeMap[Long, T] = {
```

Comment:
No, we will emit window for eventB that won't include eventA so we will emit correct value for eventA if it would be in correct order. New problem that we will introduce is that for eventB we won't emit new updated event but it can't be done well without timers :-/ But IMO it is a little bit better than it was before. Because we will emit context for eventA so we should also emit correct value for eventA. WDYT?

Reply:
ok, let's just move .to(timestamp) to stateForTimestampToRead method :)